app/components/layer-group.js

```diff
@@ -134,37 +100,49 @@ export default Ember.Component.extend(ParentMixin, ChildMixin, {
     return config;
   },
 
-  buildRangeSQL(column = '', range = [0, 1] || ['a', 'b']) {
-    let sql = this.get('config.sql');
+  buildRangeSQL(sql, column = '', range = [0, 1] || ['a', 'b']) {
+    let newSql = sql;
     let cleanRange = range;
 
     if (typeof range[0] === 'string') {
       cleanRange = cleanRange.map(step => `'${step}'`);
     }
 
-    sql += ` WHERE ${column} > ${cleanRange[0]} AND ${column} < ${cleanRange[1]}`;
+    newSql += ` WHERE ${column} > ${cleanRange[0]} AND ${column} < ${cleanRange[1]}`;
 
-    return sql;
+    return newSql;
   },
 
-  buildMultiSelectSQL(column = '', values = [0, 1] || ['a', 'b']) {
-    let sql = this.get('config.sql')[0];
-
+  buildMultiSelectSQL(sql, column = '', values = [0, 1] || ['a', 'b']) {
+    let newSql = sql;
     const valuesCleaned = values.map(value => `'${value}'`).join(',');
     if (!Ember.isEmpty(values)) {
-      sql += ` WHERE ${column} IN (${valuesCleaned})`;
+      newSql += ` WHERE ${column} IN (${valuesCleaned})`;
     }
-
-    return sql;
+    return newSql;
   },
 
   actions: {
     toggleVisibility() {
       this.toggleProperty('visible');
     },
-    updateSql(method, column, value) {
-      const sql = this[method](column, value);
-      this.set('sql', [sql]);
+    updateSql(method, sourceId, column, value) {
```

Comment:
We could get into trouble here with race conditions but this is not usually ever a long-running request so I'm not concerned.

Reply:
I'm going to open an issue on ember-mapbox-gl, I think this is a use case that can nudge the addon in a new way.